dra-evolution/pkg/api/capacity_types.go

```diff
@@ -1,170 +1,162 @@
 // expectation would a be a pool per identical collection of devices, per node.
 // It is fine to have more than one pool for a given node, for the same driver.
-type DevicePool struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   DevicePoolSpec   `json:"spec,omitempty"`
-	Status DevicePoolStatus `json:"status,omitempty"`
-}
-
-// DevicePoolSpec identifies the driver and contains the data for the pool
-// prior to any allocations.
-// NOTE: It's not clear that spec/status is the right model for this data.
-type DevicePoolSpec struct {
-	// NodeName is the name of the node containing the devices in the pool.
-	// For network attached devices, this may be empty.
+type ResourcePool struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata
 	// +optional
-	NodeName *string `json:"nodeName,omitempty"`
-
-	// Driver is the name of the DeviceDriver that created this object and
-	// owns the data in it.
-	// +required
-	Driver string `json:"driver,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-	// Attributes contains device attributes that are common to all devices
-	// in the pool.
+	// NodeName identifies the node which provides the devices
+	// if (and only if) they are local to a node. If this is empty,
+	// the devices exist outside of any specific node.
+	//
+	// NodeName and NodeSelector are mutually exclusive. If none is
+	// set, the devices are available in the entire cluster.
+	//
+	// A field selector can be used to list only ResourceSlice
+	// objects with a certain node name.
+	//
 	// +optional
-	// +listType=atomic
-	Attributes []Attribute `json:"attributes,omitempty"`
+	NodeName string `json:"nodeName,omitempty" protobuf:"bytes,2,opt,name=nodeName"`
 
-	// SharedResources are pooled resources that are shared by all devices
-	// in the pool. This is typically used when representing a
-	// partitionable device, and need not be populated otherwise.
+	// NodeSelector identifies all nodes that devices which are shared
+	// between nodes could be accessed from.
+	//
+	// NodeName and NodeSelector are mutually exclusive. If none is
+	// set, the devices are available in the entire cluster.
 	//
 	// +optional
-	// +listType=atomic
-	SharedResources []ResourceCapacity `json:"sharedResources,omitempty"`
-
-	// Devices contains the individual devices in the pool.
+	NodeSelector *v1.NodeSelector `json:"nodeSelector,omitempty"`
+
+	// ^^^^^^^^^^^^
+	// If we don't include NodeSelector in 1.31, then we need to prepare
+	// differently for adding it later. We cannot simply add it as a new
+	// field later because it would get ignored by older schedulers, which
+	// then incorrectly assume that the devices are available everywhere.
+
+	// DriverName identifies the DRA driver providing the capacity information.
+	// A field selector can be used to list only ResourceSlice
```

Comment:
We need to spec this - is it "nvidia.com/gpu" style or "gpu.nvidia.com" style or freeform?

Reply:
Driver names are DNS subdomains. I can repeat this information for each `DriverName` field.

Reply:
We also have the recommendation that they should end with a DNS domain owned by the vendor, to avoid collisions. I just noticed that we never wrote that down in the KEP - I'm taking a TODO to change that.

The current testdata uses `example.com-foozer` only because that is what the k8srm-prototype used. To me, that always looked "wrong" because "com-foozer" is not a root domain. It's technically correct, though.